include/rocksdb/utilities/transaction.h

```diff
@@ -470,7 +470,7 @@ class Transaction {
 
  protected:
   explicit Transaction(const TransactionDB* db) {}
-  Transaction() {}
+  Transaction() : log_number_(0), txn_state_(STARTED) {}
```

Comment:
@ajkr @sagar0 Should I add another enum member TransactionState::FAILED to initialize txn_state_ to FAILED in default Transaction constructor. Though in RegisterTransaction we are checking transaction state is in TransactionState::STARTED.

Reply:
@maysamyabandeh -- How would you like to initialize the txn_state_ here?

Reply:
STARTED is fine. This is always initialized to STARTED in the PessimisticTransaction constructor:
```
 56 void PessimisticTransaction::Initialize(const TransactionOptions& txn_options) {
 57   txn_id_ = GenTxnID();
 58
 59   txn_state_ = STARTED;
```